src/compiler/transformers/module/module.ts

```diff
@@ -1813,21 +1839,61 @@ namespace ts {
         }
     }
 
+    export const createBindingHelper: UnscopedEmitHelper = {
+        name: "typescript:commonjscreatebinding",
+        importName: "__createBinding",
+        scoped: false,
+        priority: 1,
+        text: `
+var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
+    if (k2 === undefined) k2 = k;
+    Object.defineProperty(o, k2, {
+        enumerable: true,
+        get: function() { return m[k]; }
+    });
+}) : (function(o, m, k, k2) {
+    if (k2 === undefined) k2 = k;
+    o[k2] = m[k];
+}));`
+    };
+
+    function createCreateBindingHelper(context: TransformationContext, module: Expression, inputName: Expression, outputName: Expression | undefined) {
+        context.requestEmitHelper(createBindingHelper);
+        return createCall(getUnscopedHelperName("__createBinding"), /*typeArguments*/ undefined, [createIdentifier("exports"), module, inputName, ...(outputName ? [outputName] : [])]);
+    }
+
+    export const setModuleDefaultHelper: UnscopedEmitHelper = {
+        name: "typescript:commonjscreatevalue",
+        importName: "__setModuleDefault",
+        scoped: false,
+        priority: 1,
+        text: `
+var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
+    Object.defineProperty(o, "default", {
+        enumerable: true,
+        value: v
+    });
+}) : function(o, v) {
+    o["default"] = v;
+});`
+    };
+
     // emit output for the __export helper function
-    const exportStarHelper: EmitHelper = {
+    const exportStarHelper: UnscopedEmitHelper = {
         name: "typescript:export-star",
-        scoped: true,
+        importName: "__exportStar",
+        scoped: false,
+        dependencies: [createBindingHelper],
+        priority: 2,
         text: `
-            function __export(m) {
-                for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
+            var __exportStar = (this && this.__exportStar) || function(m, exports) {

```

Comment:
We will need to update tslib for this as well.

Reply:
https://github.com/microsoft/tslib/pull/89

Reply:
(`tslib` already has an `__exportStar` helper, we just update it)